src/qiboml/models/ansatze.py

```diff
@@ -1,14 +1,25 @@
+from dataclasses import dataclass
+
+import numpy as np
 from qibo import Circuit, gates
 
+from qiboml.models.abstract import QuantumCircuitLayer
+
+
+@dataclass
+class ReuploadingLayer(QuantumCircuitLayer):
+
+    def __post_init__(self):
+        super().__post_init__()
+        for q in self.qubits:
+            self.circuit.add(gates.RY(q, theta=0.0))
+            self.circuit.add(gates.RZ(q, theta=0.0))
+        for i, q in enumerate(self.qubits[:-2]):
+            self.circuit.add(gates.CNOT(q0=q, q1=self.qubits[i + 1]))
+        self.circuit.add(gates.CNOT(q0=self.qubits[-1], q1=self.qubits[0]))
+
+    def forward(self, x: Circuit) -> Circuit:
+        return x + self.circuit
```

Comment:
This seems closer to function evaluation than forward derivative accumulation...

Reply:
I am not sure to get what you mean, but, indeed, the role of the forward is to process the input data, the backward takes care of evaluating the gradients. Note, anyway, that in most cases, here, the forward is not doing much other than appending a new part to the circuit. The only layers that actually execute the circuit are the decoding layers.

Reply:
I believe the misunderstanding here is the same of the previous comment. Explanation in https://github.com/qiboteam/qiboml/pull/20#discussion_r1627318244